CMakeLists.txt

```diff
@@ -174,9 +176,114 @@ add_executable(run-test262
     run-test262.c
 )
 target_compile_definitions(run-test262 PRIVATE ${qjs_defines})
-target_link_libraries(run-test262 qjs m pthread)
-if(NOT MINGW)
-    target_link_libraries(run-test262 dl)
+target_link_libraries(run-test262 ${qjs_libs})
+
+
+# Examples
+#
+
+if(BUILD_EXAMPLES AND NOT MINGW)
+    list(APPEND HELLO_OPTS
+        -fno-string-normalize
+        -fno-map
+        -fno-promise
+        -fno-typedarray
+        -fno-typedarray
+        -fno-regexp
+        -fno-json
+        -fno-eval
+        -fno-proxy
+        -fno-date
+        -fno-module-loader
+    )
+    add_custom_command(
+        OUTPUT hello.c
+        COMMAND "${CMAKE_CURRENT_BINARY_DIR}/qjsc" -e -o ./hello.c ${HELLO_OPTS} ${CMAKE_CURRENT_SOURCE_DIR}/examples/hello.js
+        DEPENDS qjsc
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+        COMMENT "Compile hello.js to a C file with bytecode embeddee"
+        SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/examples/hello.js
+    )
+    add_executable(hello
+        ${CMAKE_CURRENT_BINARY_DIR}/hello.c
+        quickjs-libc.c
+    )
+    target_include_directories(hello PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+    target_compile_definitions(hello PRIVATE ${qjs_defines})
+    target_link_libraries(hello ${qjs_libs})
+
+    list(APPEND HELLO_MODULE_OPTS
+        -fno-string-normalize
+        -fno-map
+        -fno-typedarray
+        -fno-typedarray
+        -fno-regexp
+        -fno-json
+        -fno-eval
+        -fno-proxy
+        -fno-date
+    )
+    add_custom_command(
+        OUTPUT hello_module.c
+        COMMAND "${CMAKE_CURRENT_BINARY_DIR}/qjsc" -e -o ./hello_module.c ${HELLO_MODULE_OPTS} -m ${CMAKE_CURRENT_SOURCE_DIR}/examples/hello_module.js
+        DEPENDS qjsc
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+        COMMENT "Compile hello_module.js to a C file with bytecode embeddee"
+        SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/examples/hello_module.js
+    )
+    add_executable(hello_module
+        ${CMAKE_CURRENT_BINARY_DIR}/hello_module.c
+        quickjs-libc.c
+    )
+    target_include_directories(hello_module PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+    target_compile_definitions(hello_module PRIVATE ${qjs_defines})
+    target_link_libraries(hello_module ${qjs_libs})
+
+    if(NOT MINGW)
+        add_library(fib MODULE examples/fib.c)
+        set_target_properties(fib PROPERTIES
+            PREFIX ""
+        )
```

Comment:
For my understanding: that strips the "lib" prefix, right?

Reply:
Correct!